internal/helm/repository/oci_chart_repository.go

```diff
@@ -375,17 +375,17 @@ func (r *OCIChartRepository) VerifyChart(ctx context.Context, chart *repo.ChartV
 
 	ref, err := name.ParseReference(strings.TrimPrefix(chart.URLs[0], fmt.Sprintf("%s://", registry.OCIScheme)), nameOpts...)
 	if err != nil {
-		return fmt.Errorf("invalid chart reference: %s", err)
+		return oci.VerificationResultFailed, fmt.Errorf("invalid chart reference: %s", err)
 	}
 
 	// verify the chart
 	for _, verifier := range r.verifiers {
-		if verified, err := verifier.Verify(ctx, ref); err != nil {
-			return fmt.Errorf("failed to verify %s: %w", chart.URLs[0], err)
-		} else if verified {
-			return nil
+		verified, err := verifier.Verify(ctx, ref)
+		if err != nil {
+			return verified, fmt.Errorf("failed to verify %s: %w", chart.URLs[0], err)
```

Comment:
We probably do, yeah. I've found a bug in both `oci_chart_reposiory.go` and in `OCIRepository` in that they both don't really handle multiple certificates. In our loop if an error occurs (e.g. the certificate was not used to sign the artifact) then we return straight away. However, there may still be certificates that have been set in the secret to check. It might be that we also want to check `notationgo.ErrorVerificationFailed` as well and continue so that the next certificate can be checked. If any other error occurred, then return straight away. Or perhaps it might be better to capture the `err` in an array and return that array of errors at the end of checking every certificate.

Reply:
Actually, now that I think about it, I think my approach is wrong and inefficient.  We're looping through the secret, checking if the secret ends in either `.crt` or `.pem` and then trying to verify the artifact with that one cert before moving to the next cert and trying again. It would be better to pass all certificates in one go to notation and let notation loop through the provided certs (which it already supports).

Reply:
Refactored how loading certs are handled. Changes how the verification loop works in [5e2074a](https://github.com/fluxcd/source-controller/pull/1075/commits/5e2074a49a799169ca60f3f07651a71ba9798f52).

Reply:
Originally, I did want to keep things similar to how cosign is implemented in flux but things started to get a bit too complicated so in the end I figured, why not keep things simple 😅